drivers/cache/Kconfig.nrfx

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022 Carlo Caione <ccaione@baylibre.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config CACHE_NRFX_CACHE
+	bool "NRFX cache driver"
```

Comment:
"NRFX" here is wrong, use "nRF" (or more precise as 51/2/3 won't use this, right?)

Reply:
ICACHE part can be used by older chips.

Reply:
Should I remove the `depends on` then? https://github.com/zephyrproject-rtos/zephyr/pull/67446#discussion_r1447582803

Reply:
Changed dependencies to: https://github.com/zephyrproject-rtos/zephyr/pull/67446#discussion_r1464920116